### PR TITLE
Use os.lstat to avoid following symlinks that might not exist before installation

### LINKED ISSUE #311: constructor aborts construction (computing size of symlink-file)
```
~/miniconda3/lib/python3.7/constructor/fcp.py", line 125, in check_duplicates_files
    getsize(join(extracted_package_dir, short_path)))
  File "~/miniconda3/lib/python3.7/genericpath.py", line 50, in getsize
    return os.stat(filename).st_size
FileNotFoundError: [Errno 2] No such file or directory: '~/.conda/constructor/linux-64/gsl-2.5-h294904e_1/lib/libgslcblas.so.0'
```
environment:
<details>
  <summary>Click to expand!</summary>

```_libgcc_mutex             0.1                        main  
anaconda-client           1.7.2                      py_0    conda-forge
attrs                     19.3.0                     py_0    conda-forge
beautifulsoup4            4.8.1                    py37_0    conda-forge
blinker                   1.4                        py_1    conda-forge
bzip2                     1.0.8                h516909a_1    conda-forge
ca-certificates           2019.9.11            hecc5488_0    conda-forge
certifi                   2019.9.11                py37_0    conda-forge
cffi                      1.13.2           py37h8022711_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
clyent                    1.2.2                      py_1    conda-forge
conda                     4.7.12                   py37_0    conda-forge
conda-build               3.18.11                  py37_0    conda-forge
conda-forge-pinning       2019.11.01                    1    conda-forge
conda-pack                0.4.0                      py_0    conda-forge
conda-package-handling    1.6.0            py37h516909a_0    conda-forge
conda-smithy              3.6.2                      py_0    conda-forge
constructor               3.0.0                    py37_0    conda-forge
cryptography              2.8              py37h72c5cf5_0    conda-forge
curl                      7.65.3               hf8cf82a_0    conda-forge
decorator                 4.4.1                      py_0    conda-forge
deprecated                1.2.6                      py_0    conda-forge
expat                     2.2.5             he1b5a44_1004    conda-forge
filelock                  3.0.10                     py_0    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
git                       2.23.0          pl526hce37bd2_2    conda-forge
gitdb2                    2.0.6                      py_0    conda-forge
gitpython                 3.0.4                      py_0    conda-forge
glob2                     0.7                        py_0    conda-forge
greenlet                  0.4.15           py37h516909a_0    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
idna                      2.8                   py37_1000    conda-forge
importlib_metadata        0.23                     py37_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isodate                   0.6.0                      py_1    conda-forge
jinja2                    2.10.3                     py_0    conda-forge
jsonschema                3.1.1                    py37_0    conda-forge
jupyter_core              4.5.0                      py_0    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libarchive                3.4.0                hb0c0d93_1    conda-forge
libcurl                   7.65.3               hda55be3_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0  
libiconv                  1.15              h516909a_1005    conda-forge
liblief                   0.9.0                hf8a498c_1    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libxml2                   2.9.10               hee79883_0    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
lzo                       2.10              h14c3975_1000    conda-forge
markupsafe                1.1.1            py37h516909a_0    conda-forge
more-itertools            7.2.0                      py_0    conda-forge
msgpack-python            0.6.2            py37hc9558a2_0    conda-forge
msrest                    0.6.10                     py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
neovim                    0.3.1                    py37_0    conda-forge
oauthlib                  3.0.1                      py_0    conda-forge
openssl                   1.1.1d               h516909a_0    conda-forge
patchelf                  0.10                 he1b5a44_0    conda-forge
pcre                      8.43                 he1b5a44_0    conda-forge
perl                      5.26.2            h516909a_1006    conda-forge
pip                       19.3.1                   py37_0    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
psutil                    5.6.5            py37h516909a_0    conda-forge
py-lief                   0.9.0            py37he1b5a44_1    conda-forge
pycosat                   0.6.3           py37h516909a_1002    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pycrypto                  2.6.1           py37h516909a_1003    conda-forge
pygithub                  1.43.6                   py37_0    conda-forge
pyjwt                     1.7.1                      py_0    conda-forge
pynvim                    0.3.2                    py37_0    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pyrsistent                0.15.5           py37h516909a_0    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-libarchive-c       2.9                      py37_0    conda-forge
pytz                      2019.3                     py_0    conda-forge
pyyaml                    5.1.2            py37h516909a_0    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
requests                  2.22.0                   py37_1    conda-forge
requests-oauthlib         1.2.0                      py_0    conda-forge
ripgrep                   11.0.2               h516909a_3    conda-forge
ruamel.yaml               0.16.5           py37h516909a_1    conda-forge
ruamel.yaml.clib          0.2.0            py37h516909a_0    conda-forge
ruamel_yaml               0.15.71         py37h14c3975_1000    conda-forge
setuptools                41.6.0                   py37_1    conda-forge
six                       1.13.0                   py37_0    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
soupsieve                 1.9.4                    py37_0    conda-forge
sqlite                    3.30.1               hcee41ef_0    conda-forge
tk                        8.6.9             hed695b0_1003    conda-forge
toolz                     0.10.0                     py_0    conda-forge
tqdm                      4.36.1                     py_0    conda-forge
traitlets                 4.3.3                    py37_0    conda-forge
urllib3                   1.25.6                   py37_0    conda-forge
vsts-python-api           0.1.22                     py_0    conda-forge
wheel                     0.33.6                   py37_0    conda-forge
wrapt                     1.11.2           py37h516909a_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.1.7             h14c3975_1001    conda-forge
zipp                      0.6.0                      py_0    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
zstd                      1.4.3                h3b9ef0a_0    conda-forge
```
</details>

### PR DESCRIPTION
**Pull request by hmaarrfk**
Fixes #311

Alternative to https://github.com/conda/constructor/pull/341
Closes https://github.com/conda/constructor/pull/341

Not sure how to add a changelog notice.

### CODE REVIEW

File: constructor/fcp.py [hunk 1]
```diff
@@ -21,6 +21,16 @@
...
 
 
+def getsize(filename):
+    """Return the size of a file, reported by os.lstat as opposed to os.stat."""
+    # Symlinks might be reported as "not found" if they are provided by a
+    # package's dependencies
+    # We use lstat to obtain the size of the symlink, as opposed to the
+    # size of the file it points to
+    # https://github.com/conda/constructor/issues/311
+    return os.lstat(filename).st_size
```
Comment (mcg1969) [inline constructor/fcp.py:31]: Nice. Can we confirm that this works across all supported operating systems?

Comment (hmaarrfk) [reply constructor/fcp.py:31]: > On platforms that do not support symbolic links, this is an alias for stat().

Comment (hmaarrfk) [reply constructor/fcp.py:35]: I added the same information to the source. I hope that is enough.

Comment (mcg1969) [reply constructor/fcp.py:35]: Thanks!

**Pull request merged**